courses/views_tests.py

```diff
@@ -48,3 +54,92 @@ def test_doesnt_list_courses_from_unlive_programs(self):
         resp = self.client.get(reverse('course-list'))
 
         assert len(resp.json) == 0
+
+
+class ProgramEnrollmentTests(ESTestCase, APITestCase):
+    """Tests for the ProgramEnrollment API"""
+
+    @classmethod
+    def setUpTestData(cls):
+        super(ProgramEnrollmentTests, cls).setUpTestData()
+
+        cls.user1 = UserFactory.create()
+        cls.user2 = UserFactory.create()
+        cls.program1 = ProgramFactory.create(live=True)
+        cls.program2 = ProgramFactory.create(live=True)
+        cls.program3 = ProgramFactory.create(live=True)
+        for program in (cls.program1, cls.program2,):
+            ProgramEnrollmentFactory(
+                user=cls.user1,
+                program=program,
+            )
+        cls.url = reverse('user_program_enrollments')
+
+    def setUp(self):
+        super(ProgramEnrollmentTests, self).setUp()
+        for program in (self.program1, self.program2,):
+            ProgramEnrollmentFactory(
+                user=self.user1,
+                program=program,
+            )
+        self.client.force_login(self.user1)
+
+    def tearDown(self):
+        super(ProgramEnrollmentTests, self).tearDown()
+        ProgramEnrollment.objects.all().delete()
+
+    def test_anonymous(self):
+        """Anonymous user cannot access the endpoint"""
+        self.client.logout()
+        resp = self.client.get(self.url)
+        assert resp.status_code == status.HTTP_403_FORBIDDEN
+
+    def test_no_enrollments(self):
+        """Requests for a user with no program enrollments result in an empty list"""
+        self.client.logout()
+        self.client.force_login(self.user2)
+        resp = self.client.get(self.url)
+        assert resp.status_code == status.HTTP_200_OK
+        assert resp.data == []
+
+    def test_enrollments(self):
+        """Only the programs where the user is enrolled in are returned"""
+        resp = self.client.get(self.url)
+        assert resp.status_code == status.HTTP_200_OK
+        expected_enrollments_ids = [
+            program.pk for program in (self.program1, self.program2,)
+        ]
+        for enr in resp.data:
+            assert enr.get('id') in expected_enrollments_ids
+
+    def test_crete_no_program_id(self):
+        """Missing mandatory program_id parameter"""
+        resp = self.client.post(self.url, {}, format='json')
+        assert resp.status_code == status.HTTP_400_BAD_REQUEST
```

Comment:
Can you add an assert for each unhappy case that the number of `ProgramEnrollment`s didn't change?